controllers/storagecluster/cephcluster.go

```diff
@@ -1198,3 +1218,30 @@ func getMonitoringClient() (*monitoringclient.Clientset, error) {
 	}
 	return client, nil
 }
+
+// getIPFamilyConfig checks for a Single Stack IPv6 or a Dual Stack cluster
+func getIPFamilyConfig(c client.Client) (rookCephv1.IPFamilyType, bool, error) {
+	isIPv6 := false
+	isIPv4 := false
+	networkConfig := &configv1.Network{}
+	err := c.Get(context.TODO(), types.NamespacedName{Name: "cluster", Namespace: ""}, networkConfig)
+	if err != nil {
+		return "", false, fmt.Errorf("could not get network config details. %v", err)
+	}
+
+	for _, cidr := range networkConfig.Status.ClusterNetwork {
+		if strings.Count(cidr.CIDR, ":") < 2 {
+			isIPv4 = true
+		} else if strings.Count(cidr.CIDR, ":") >= 2 {
+			isIPv6 = true
+		}
+	}
+
+	if isIPv4 && isIPv6 {
+		return "", true, nil
+	} else if isIPv6 && !isIPv4 { // IPv6 single stack cluster
```

Comment:
```suggestion
} else if isIPv6 {
```
The above is enough right? we don't need to check isIPv4 here again

Reply:
Right. First condition should be enough. Update the code.